src/main/java/net/dv8tion/jda/internal/entities/EntityBuilder.java

```diff
@@ -118,7 +118,8 @@ public SelfUser createSelfUser(DataObject self)
                 .setName(self.getString("username"))
                 .setDiscriminator(self.getString("discriminator"))
                 .setAvatarId(self.getString("avatar", null))
-                .setBot(self.getBoolean("bot"));
+                .setBot(self.getBoolean("bot"))
+                .setSystem(self.getBoolean("system", false));
```

Comment:
Isn't this always false?

Reply:
It's marked as optional on the docs so I figured it was best to add a fallback.

Reply:
Just checked, it's undefined for normal users: `{"id": "86699011792191488", "username": "Minn", "avatar": "f3bc77b1e276fba2e1485a915141564b", "discriminator": "6688", "public_flags": 512}`, although so is bot, so I guess `getBoolean` just decodes undefined as false.

Reply:
I was more referring to self user never being system=true